Source/VMware.PSDesiredStateConfiguration/Classes/DscConfigurationFileParser.ps1

```diff
@@ -169,4 +179,65 @@ class DscConfigurationFileParser {
 
         return $dscConfigurations
     }
+
+    <#
+    .DESCRIPTION
+
+    Creates a ScriptBlock for the specified script content with the passed script parameters.
+    #>
+    hidden [ScriptBlock] CreateScriptBlockWithParameters([string] $scriptContent, [System.Collections.Hashtable] $parameters) {
+        $modifiedParameters = @{}
+        foreach ($key in $parameters.Keys) {
+            if ($parameters.$key -is [array]) {
+                <#
+                    Array parameters should be enclosed in quotes because if they aren't
+                    the ToString() method is called which results in System.Object[] as value of the
+                    parameter. With quotes the original value of the array parameter is kept.
+                #>
+                $arrayParameterAsString = "@("
+                for ($i = 0; $i -lt $parameters.$key.Length; $i++) {
+                    $arrayElement = $parameters.$key[$i]
+
+                    # String values in the array should be enclosed in quotes.
+                    if ($arrayElement -is [string]) {
+                        $arrayParameterAsString += "'"
+                        $arrayParameterAsString += $arrayElement
+                        $arrayParameterAsString += "'"
+                    }
+                    else {
+                        $arrayParameterAsString += $arrayElement
```

Comment:
This won't work for objects that don't have conversion object->tostring->toobject
E.g. hashtables will fail.

Array of array is not handled as well